app/forms/openid_connect_logout_form.rb

```diff
@@ -13,24 +14,26 @@ class OpenidConnectLogoutForm
 
   RANDOM_VALUE_MINIMUM_LENGTH = OpenidConnectAuthorizeForm::RANDOM_VALUE_MINIMUM_LENGTH
 
-  validates :id_token_hint, presence: true
+  validates :id_token_hint, presence: true, unless: :reject_id_token_hint?
   validates :post_logout_redirect_uri, presence: true
   validates :state, presence: true, length: { minimum: RANDOM_VALUE_MINIMUM_LENGTH }
 
-  validate :validate_identity
+  validate :client_id_present, if: :reject_id_token_hint?
+  validate :validate_identity, unless: :reject_id_token_hint?
+  validate :no_id_token_hint_present, if: :reject_id_token_hint?
 
   def initialize(params)
     ATTRS.each do |key|
       instance_variable_set(:"@#{key}", params[key])
     end
 
-    @identity = load_identity
+    @identity = load_identity unless reject_id_token_hint?
   end
 
   def submit
     @success = valid?
 
-    identity.deactivate if success
+    identity.deactivate if (success && !reject_id_token_hint?)
```

Comment:
I think we still want to find a way to deactivate the identity, remove the UUID etc if possible

Could we find the identity like:
```ruby
current_user.identities.find_by(service_provider: client_id)&.deactivate
```

Reply:
@zachmargolis  good call - that only works if the user already has an active session, but honestly that's the only case where it matters if we deactivate the identity. I'm assuming we end up cleaning up that `session_uuid` if the session times out anyway?